bin/spack.bat

```diff
@@ -96,20 +96,23 @@ set str_subcommand=%_sp_subcommand:"='%
 set str_flags=%_sp_flags:"='%
 set str_args=%_sp_args:"='%
 if "%str_subcommand%"=="ECHO is off." (set "_sp_subcommand=")
+if "%str_subcommand%"=="ECHO is on." (set "_sp_subcommand=")
 if "%str_flags%"=="ECHO is off." (set "_sp_flags=")
+if "%str_flags%"=="ECHO is on." (set "_sp_flags=")
 if "%str_args%"=="ECHO is off." (set "_sp_args=")
+if "%str_args%"=="ECHO is on." (set "_sp_args=")
 del subcmd
 del flags
 del args
 
 :: Filter out some commands. For any others, just run the command.
-if "%_sp_subcommand%" == "cd" (
+if %_sp_subcommand% == "cd" (
```

Comment:
Is this to avoid issues if `%_sp_subcommand%` contains a space?

Reply:
All the issues I fixed here resulted from the fact that _sp_subcommand and _sp_args were already between double quotes. So for instance the first test lower would compare ""env"" with "env" that would fail.

Reply:
Sorry did we add the "first" set of quotes ourselves? Or is this for cases where args are quoted explicitly by the user on the command line?

Also, given that, how does the following work:

```
set str_subcommand=%_sp_subcommand:"='%
if "%str_subcommand%"=="ECHO is on." (set "_sp_subcommand=")
```

It looks like "if" would never be true (if as you say _sp_subcommand contains " characters, then `str_subcommand` would have `'` characters and it looks like we'd be comparing `"'ECHO is on'" == "ECHO is on."`)

Reply:
The ECHO is on/off is different. That comes without double quotes from doing echo on an undefined variable on line 88.

Reply:
If I understand, in that case defining `str_subcommand` is unnecessary and 

```
if "%str_subcommand%"=="ECHO is off." (set "_sp_subcommand="
```

for example could be just testing `sp_subcommand`

Reply:
That is correct. Good catch! I pushed another commit that gets rid of those variables.